### PR TITLE
Update revision date upon cipher restore

### PR DESCRIPTION
**Pull request by MGibson1**
# Overview

https://app.asana.com/0/1169444489336079/1199337616674667/f
related to bitwarden/web/issues/748

Cipher edit revision date validation is failing for ciphers which were restored from the trash. This is because restores update revisionDate on the server side, but do not send an updated cipher back to the client.

These changes force a call to the server after restore to get the new revisionDate of the cipher.

# Files Changed
* **api.service.ts**: Change signature to expect response from restore calls. (bitwarden/server#1072)
* **cipher.service.ts**: Update local cipher db to match ciphers returned from restore calls

### CODE REVIEW

File: src/services/cipher.service.ts
Comment (MGibson1) [inline src/services/cipher.service.ts:915]: this feels hacky to me. Should we just change the server signature to return the updated ciphers?

Comment (cscharf) [reply src/services/cipher.service.ts:915]: I like that much better... shouldn't hurt reverse compatibility in that way since the client can ignore returned JSON (older client versions) but newer client versions can take advantage of it. Any time we're hitting an API (or database) in a loop it makes my skin crawl a bit

File: src/abstractions/cipher.service.ts
Comment (MGibson1) [reply src/abstractions/cipher.service.ts:56]: Nope, it looks like this could be removed from the interface and made private in the service and no client would care.

Shall I do that?

File: src/services/cipher.service.ts
Comment (MGibson1) [reply src/services/cipher.service.ts]: The server changes are in bitwarden/server#1072

**Review by cscharf — APPROVED** [review]
Looks good, may just want to wait until server update is approved/merged before merging this one.

**Pull request merged**
